src/NLPModels.jl

```diff
@@ -31,8 +31,13 @@ function reset!(counters :: Counters)
   return counters
 end
 
+"Reset evaluation count in `nlp`"
+function reset!(nlp :: AbstractNLPModel)
+  reset!(nlp.counters)
+  return nlp
+end
+
```

Comment:
Doesn't this allow a user to define an object of type `AbstractNLPModel` and call `reset!()`, which would error out because there's no field named `counters`?